webdriver/newsession/invalid_capabilities.py

```diff
@@ -0,0 +1,89 @@
+#META: timeout=long
+
+import pytest
+from webdriver import error
+
+from util.newsession import new_session
+from conftest import platform_name, product, flatten
+
+
+@pytest.mark.parametrize("value", [None, 1, "{}", []])
+def test_invalid_capabilites(_function_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session(_function_session, {"capabilities": value})
+
+
+@pytest.mark.parametrize("value", [None, 1, "{}", []])
+def test_invalid_always_match(_function_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session(_function_session, {"capabilities": {"alwaysMatch": value}})
+
+
+@pytest.mark.parametrize("value", [None, 1, "[]", {}])
+def test_invalid_first_match(_function_session, value):
+    with pytest.raises(error.InvalidArgumentException):
+        new_session(_function_session, {"capabilities": {"firstMatch": value}})
+
+
+invalid_data = [
+    ("acceptInsecureCerts", [1, [], {}, "false"]),
+    ("browserName", [1, [], {}, False]),
+    ("browserVersion", [1, [], {}, False]),
+    ("platformName", [1, [], {}, False]),
+    ("pageLoadStrategy", [1, [], {}, False, "invalid", "NONE", "Eager", "eagerblah", "interactive",
+                          " eager", "eager "]),
+    ("proxy", [1, [], "{}", {"proxyType": "SYSTEM"}, {"proxyType": "systemSomething"},
```

Comment:
I don't think that's right: https://w3c.github.io/webdriver/webdriver-spec.html#dfn-deserialize-as-a-proxy There's no case insensitivity there.